docs/README.md

```diff
@@ -15,52 +21,61 @@ You can view the latest successful build here: https://docs.aztec.network
 
 The .md files in the `docs/` directory are the docs. See the [Docusaurus website](https://docusaurus.io/docs/docs-introduction) for the full documentation on how to create docs and to manage the metadata.
 
-## Contributing
-
-We welcome contributions from the community. Please review our [contribution guidelines](CONTRIBUTING.md) for more information.
-
-## Docusaurus
-
-This website is built using [Docusaurus 2](https://docusaurus.io/), a modern static website generator.
-
 ### Installation
 
+To install the dependencies and dev dependencies, run:
+
 ```
 $ yarn
 ```
 
 ### Development
 
-#### Locally
+Aztec docs pull some code from the rest of the repository. This allows for great flexibility and maintainability. Some documentation is also autogenerated.
+
+For that reason, there's a preprocessing step. You can run that step ad-hoc with `yarn preprocess` or `yarn preprocess:dev` if you want it to stay running and watching for changes.
 
-To build; serve to `localhost:3000`; and watch for changes:
+This step does the following:
+- Pulls the code from the source files using the `#include` macros explained below.
+- Autogenerates documentation using the scripts in the `src` file.
+- Puts the final documentation in a `processed-docs` folder.
+
+> [!NOTE]
+> You likely want to benefit from webpack's hot reload, which allows you to immediately see your changes when you develop on the docs. For this reason, the `yarn dev` commands will add the `ENV=dev` environment variable, which makes docusaurus serve the `docs folder` instead of the `processed docs`.
```

Comment:
Sounds great for visual changes.
Apart from URLs, what is in processed docs that might need to be tested? Might just need a comment with what is not accurate between `docs` vs `processed_docs`.
Saw that CI is building `staging` docs :+1:

Reply:
Exactly, it just adds the autogenerated stuff